Foundation/FileHandle.swift

```diff
@@ -109,7 +109,7 @@ open class FileHandle : NSObject, NSSecureCoding {
         
```

Comment:
For clarity, can we change this:

`if (0 == total) { } if (total > 0) { }` to an if-else?

Just so we make sure we can't exit this function without either a free or data creation with `.free`.